pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1468,23 +1468,29 @@ func (c *Cloud) AttachDisk(diskName string, instanceName string, readOnly bool)
 	// See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
 	ec2Device := "/dev/xvd" + string(mountDevice)
 
-	if !alreadyAttached {
-		request := &ec2.AttachVolumeInput{
-			Device:     aws.String(ec2Device),
-			InstanceId: aws.String(awsInstance.awsID),
-			VolumeId:   aws.String(disk.awsID),
-		}
+	request := &ec2.AttachVolumeInput{
+		Device:     aws.String(ec2Device),
+		InstanceId: aws.String(awsInstance.awsID),
+		VolumeId:   aws.String(disk.awsID),
+	}
 
-		attachResponse, err := c.ec2.AttachVolume(request)
-		if err != nil {
+	attachResponse, err := c.ec2.AttachVolume(request)
```

Comment:
Even if alreadyAttached returns true, what is the reason you still want to trigger AttchVolume operation?